airflow/operators/python_operator.py

```diff
@@ -86,6 +87,38 @@ def __init__(
         if templates_exts:
             self.template_ext = templates_exts
 
+    def __deepcopy__(self, memo):
+        """
+        Hack sorting double chained task lists by task_id to avoid hitting
+        max_depth on deepcopy operations.
+        """
+        sys.setrecursionlimit(5000)  # TODO fix this in a better way
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+
+        # since we won't mutate the arguments, we should just do the shallow copy
+        # there are some cases we can't deepcopy the objects(e.g protobuf).
+        shallow_copy = {'python_callable', 'op_kwargs'}
+        skip_deepcopy = {'user_defined_macros',
```

Comment:
To me there are just two types of arguments here, the ones we deepcopy and the ones we shallow copy. We can assume deepcopy is the default and only have a set of `shallow_copy` attrs.

Reply:
thanks. This is better. pr updated.